### ISSUE TITLE
Hide completed tasks in checklists

### ISSUE THREAD

**Issue by torkelsson** [nextcloud/deck#2931]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Is your feature request related to a problem? Please describe.**
When having a checklist with many tasks, it is sometimes very easy to miss an unchecked task in the middle of checked tasks, therefore a nice addition would be to be able to hide/show checked tasks, primarily in rendered "View Description" mode but also (if at all possible) in "Edit Description".

When coming from Trello and trying to migrate to Deck this is a feature that are used a lot. 

**Describe the solution you'd like**
An option to hide/show completed/checked tasks of a checklist.  Primarily when in markdown rendered mode ("View Description") but if possible also when in Editing mode.

**Describe alternatives you've considered**
An option would be to move the completed tasks to another checklist. That would require a lot more manual work though and forfeits the usefulness of checklists. Moving tasks between different lists also increases the risk of changing to order of the original tasks (if that matters).

**Comment by stefan-niedermann** [comment]
As already mentioned in the corresponding [feature request for the Deck Android app](https://github.com/stefan-niedermann/nextcloud-deck/issues/911) i want to explain why i downvoted this feature request.

1. Automatically modifying the content as requested (like moving items around to other lists or the top or bottom) might match for *some* use cases but be bad and unexpected for other use cases. Since we don't know how people are using the app, we can only make assumptions and changing the order of items automatically might **change the semantic** of a list. A simple example is a task list where the tasks depend on other tasks and the order is important. We simply don't *know* the intention of the author and therefore should not manipulate the list in my opinion.

2. Also, i think the benefit of the feature request is low for the majority of the users and therefore does not justify an own setting.
    >  it is sometimes very easy to miss an unchecked task in the middle of checked tasks

    So taking an actual advantage of this logic requires users who make use of task lists and adding so many items, that it doesn't fit into one screen *and* the user to not scroll :thinking: ? How often does this really happen? Maybe a solution might be to rethink your structure / workflow and pull the tasks out into own cards (references Sub-cards #2103)?

3. I fear the maintenance of this kind of automatism *and* a setting to toggle it, will cost too much developer resources. We need to deal with a documentation and handling questions of users which do not understand the mechanism while we have little resources to bring other features for potentially more users on the road.
4. "Trello has this too" is a hint, but not a reason. For example 3. won't affect them as much as us. :wink: 
5. From the server apps point of view i think you want to [switch to Nextcloud Text](https://github.com/nextcloud/deck/pull/1663) in the future where edit & preview views will be merged. This needs to be considered and doing some magic while editing will be even more disturbing and confusing behavior.

As also stated in the Deck Android issue, **i will not block this and probably also implement this in case the server app does it**, i just wanted to elaborate a little more why i don't think that this is a good idea.

**Comment by torkelsson** [comment]
Sorry for being unclear. I was not thinking on actually modifying the list, but more of a way to do a different presentation of the checklist when in "View Description" mode, i.e a button next to the "pen" to hide/show completed items (or at the beginning of the list).

With a "setting" to switch the behavior I do see an extra maintenance burden, but not as much with a button. As I don't  really know anything about Deck I could off course be completely wrong though. 

A use-case would be a shopping list (I know there are probably better apps for that, but hopefully you will get the idea) or a list with many small tasks that does not justify their own cards. In the case of a shopping list the list is probably in the order you are going to pick them up at the store. When att the end of the list you want to quickly see if you forgot something. Imo the easiest and quickest way to do that is to hide what you have picked up and only show what you have left. 

I agree that just because something else does have a feature is not a reason to add it to Deck, though in this case I think that it will be a feature to make the move from Trello (or other Kanban boards) easier and would add an extra value to Deck.

**Comment by torkelsson** [comment]
An additional comment: Perhaps in the future checklists/tasks will be more of a first class citizen in Deck, having their own tab, just list attachments, comments etc. On the other hand then it might be even closer to sub-cards?

**Comment by stefan-niedermann** [comment]
No, i think i got your intention :slightly_smiling_face:, you don't want to modify the actual content but the visual representation. 

And this is just one more issue i can see: Between edit and view mode will be a difference, while they should (in my opinion) be as similar as possible to provide an easy orientation when switching between them. Adding 3 items to a shopping list and finding them in the super market at a different position compared to where i *put* it, would confuse me.

> Imo the easiest and quickest way to do that is to hide what you have picked up and only show what you have left.

This is correct, in the use case of a shopping list. If you plan some BBQ and want to check who of your guests wants meat and who wants vegetables, it will not fit. Because we assume to know the *semantic* of the task list, but it will not be right in each scenario.

Well, in the rare cases where i have so many tasks that i don't see at the first look if they all are checked or not, i use the counter at the card to see whether a task is missing and then just have a look which one it is:

![Bildschirmfoto am 2021-04-01 um 13 09 33](https://user-images.githubusercontent.com/4741199/113285668-85852f00-92eb-11eb-9c76-45a023a3cbc4.png)

> first class citizen in Deck

I value the freedom which Markdown gives me while creating cards and personally wouldn't like a separation of tasks, which then no longer are embedded within flow text and don't have any context anymore, but this is quite personal taste, i guess. Integration and benefits (like the counter above): :+1:, splitting them out of the existing markdown description: :-1: 

Looking forward for more opinions on this topic and a decision by @juliushaertl :slightly_smiling_face:
